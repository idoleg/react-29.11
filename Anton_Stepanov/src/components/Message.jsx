import React from "react";

export const Message = (props) => <div><b>{props.name}:</b> {props.content}</div>;