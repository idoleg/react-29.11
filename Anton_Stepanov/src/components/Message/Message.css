.message {
    background-color: lightblue;
    border-radius: 10px;
    padding: 10px;
    margin: 5px 0;
    width: 30%;
}

.message-person {
    background-color: lightgoldenrodyellow;
    align-self: flex-end;
    margin-right: 20px;
}