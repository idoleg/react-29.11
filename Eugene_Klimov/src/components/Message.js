import React from 'react';
import PropTypes from 'prop-types';

export const Message = (props) =>
  <div className="user-item">
    <span className="user-name">{props.name}: </span>
    <span className="user-content">{props.content}</span>
  </div>;

Message.propTypes = {
  name: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
};
