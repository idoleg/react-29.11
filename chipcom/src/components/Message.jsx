import React from 'react';

export default class Message extends React.Component {
    
    render() {
        return (
            <div>
                <b>{this.props.name}: </b>
                {this.props.text}
            </div>
        )
    }
}