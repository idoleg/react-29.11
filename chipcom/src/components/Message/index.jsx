import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const dateFormat = require('dateformat');

class Message extends React.Component {
    static propTypes = {
        text: PropTypes.string.isRequired,
        sender: PropTypes.string.isRequired,
    };
    
    render() {
        dateFormat.i18n = {
            dayNames: [
                'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat',
                'воскресенье', 'понедельник', 'вторник', 'среда', 'четверг', 'пятница', 'суббота'
            ],
            monthNames: [
                'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
                'января', 'февраля', 'марта', 'апреля', 'мая', 'июня', 'июля', 'августа', 'сентября', 'октября', 'ноября', 'декабря'
            ],
            timeNames: [
                'a', 'p', 'am', 'pm', 'A', 'P', 'AM', 'PM'
            ]
        };

        const strDate = dateFormat(this.props.created, 'd mmmm yyyy HH:MM');

        return <div
            className="message"
            style={ { alignSelf: this.props.sender === 'bot' ?
                'flex-start' : 'flex-end' } }
            >
            <div className="message-sender">{ this.props.sender }</div>
            <div className="message-date">Cоздан: { strDate }</div>
            <div>{ this.props.text }</div>
        </div>
    }
}

export default Message;