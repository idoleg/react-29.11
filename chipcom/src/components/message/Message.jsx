import React from 'react';
import PropTypes from 'prop-types';
import './message.css';

class Message extends React.Component {
    static propTypes = {
        text: PropTypes.string.isRequired,
        sender: PropTypes.string.isRequired,
    };

    render() {
        return <div
            className="message"
            style={ { alignSelf: this.props.sender === 'bot' ?
                'flex-start' : 'flex-end' } }
            >
            <div className="message-sender">{ this.props.sender }</div>
            <div>{ this.props.text }</div>
        </div>
    }
}

export default Message;