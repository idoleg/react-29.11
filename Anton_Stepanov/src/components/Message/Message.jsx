import React, {Component} from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import("./Message.css");

export const messageType = {
    name: PropTypes.string,
    content: PropTypes.string.isRequired
};

export class Message extends Component {
    static propTypes = messageType;

    render(){
        const {name, content} = this.props;
        const msgClass = classNames("message", {
            "message-person": name !== "Robot"
        });
        // const style = {alignSelf: name === 'Robot' ? 'flex-start' : 'flex-end'}
        return (
            <div className={msgClass}><b>{name}:</b> {content}</div>
        );
    }
}