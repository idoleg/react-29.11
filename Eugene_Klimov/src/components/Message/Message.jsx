import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {botName} from '../../utils/constants';
import DeleteOutlinedIcon from '@material-ui/icons/DeleteForeverOutlined';
import './Message.sass';
import {bindActionCreators} from 'redux';
import {deleteMessage} from '../../actions/messageActions';
import {connect} from 'react-redux';

const messageType = {
  author: PropTypes.string,
  content: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  id: PropTypes.number,
  chatId: PropTypes.string,
  deleteMessage: PropTypes.func,
};

export class Message extends Component {
  static propTypes = messageType;

  handleDeleteMessage = () => {
    confirm('Действительно удалить сообщение?') ?
      this.props.deleteMessage(this.props.chatId, this.props.id) : null;
  };

  render() {
    const {author, content, date} = this.props;
    return (
      <div className={author === botName ? 'bot-item' : 'user-item'}>
        <div className='user-name-trash'>
          <span className='user-name'>{author || 'Anonymous'}</span>
          <DeleteOutlinedIcon className='delete-icon' viewBox='0 0 36 36'
                              onClick={this.handleDeleteMessage}
          />
        </div>
        <span className='user-date'>{date}</span>
        <span className='user-content'>{content}</span>
      </div>
    );
  }
}

const mapStateToProps = ({}) => ({});

const mapDispatchProps = (dispatch) =>
  bindActionCreators({deleteMessage}, dispatch);

export default connect(mapStateToProps, mapDispatchProps)(Message);
