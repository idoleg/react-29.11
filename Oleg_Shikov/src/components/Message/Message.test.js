import React from "react";
import renderer from "react-test-renderer";
import {Message} from "./Message";
import {shallow} from "enzyme";

describe("<Message>", () => {

    it("renders our content", () => {
        const props = {name: "My user", content: "My content"};

        const element = renderer.create(<Message {...props} />).toJSON();
        expect(element).toMatchSnapshot();

    })


    it("message with name", () => {
        const props = {content: "My content"};

        const element = shallow(<Message {...props} />);

        expect(element.contains(<strong>Anonim:</strong>)).toBe(true)
        element.setProps({ name: 'My user' });
        expect(element.contains(<strong>My user:</strong>)).toBe(true)
    })
});