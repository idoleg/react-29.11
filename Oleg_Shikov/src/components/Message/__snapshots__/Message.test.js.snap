// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message> renders our content 1`] = `
<div
  className="message message-person"
>
  <strong>
    My user
    :
  </strong>
   
  My content
</div>
`;
